exec/internal/compile/scanner.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package compile
+
+import (
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+)
+
+type scanner struct {
+	supportedOpcodes map[byte]bool
+}
+
+// InstructionMetadata describes a bytecode instruction.
+type InstructionMetadata struct {
+	Op    byte
+	Start int
+	Size  int
+}
+
+// CompilationCandidate describes a range of bytecode that can
+// be translated to native code.
+type CompilationCandidate struct {
+	// Bytecode index of the first opcode.
+	Beginning uint
+	// Bytecode index of the last byte in the instruction.
+	End uint
+	// InstructionMeta index of the first instruction.
+	StartInstruction int
+	// InstructionMeta index of the last instruction.
+	EndInstruction int
+	// Metrics about the instructions between first & last index.
+	Metrics *Metrics
```

Comment:
it seems we always expect `Metrics` to be there.
why don't we make it `Metrics` instead of `*Metrics` ?